src/main/scala/com/gossiperl/client/GossiperlProxy.scala

```diff
@@ -0,0 +1,131 @@
+package com.gossiperl.client
+
+import akka.actor.{ActorRef, ActorLogging, Actor}
+import akka.util.Timeout
+import com.gossiperl.client.SupervisorProtocol.SupervisorAction
+import com.gossiperl.client.actors.ActorRegistry
+import com.gossiperl.client.serialization.{Serializer, DeserializeResult, CustomDigestField}
+
+import scala.concurrent.{Promise, Future}
+import scala.util.{Try, Failure, Success}
+
+import concurrent.duration._
+
+object GossiperlProxyProtocol {
+  case class Connected( configuration: OverlayConfiguration )
+  case class Disconnected( configuration: OverlayConfiguration )
+  case class Event( configuration: OverlayConfiguration, eventType: String, member: Any, heartbeat: Long )
+  case class SubscribeAck( configuration: OverlayConfiguration, eventTypes: Seq[String] )
+  case class UnsubscribeAck( configuration: OverlayConfiguration, eventTypes: Seq[String] )
+  case class ForwardAck( configuration: OverlayConfiguration, replyId: String )
+  case class Forward( configuration: OverlayConfiguration, digestType: String, binaryEnvelope: Array[Byte], envelopeId: String )
+  case class Error( configuration: OverlayConfiguration, error: Throwable )
+
+  case class ShutdownRequest( p: Promise[ActorRef] )
+}
+
+class GossiperlProxy( val configuration: OverlayConfiguration, p: Promise[GossiperlProxy] ) extends ActorRegistry with ActorLogging {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+  implicit val timeout = Timeout(1 seconds)
+
+  // Yes, we do want it like this. We want the proxy to be available as an instance to the caller so we can simply forward events to it.
+  // Proxy is used from outside of the actor system.
+  p.success(this)
```

Comment:
This has some background to it. Initially I wanted the `GossiperlProxy` class to use the actor(new Act{}) from the DSL package. I was attempting something like this (in the supervisor):

    case Connect( ... ) =>
      new GossiperlProxy( ..., context.system )

To have the same system available in the proxy object (proxy object is used from outside of the actor system). actor(new Act{}) expects an implicit system, which I was setting in GossiperlProxy like this:

    class GossiperlProxy( ..., actorSystem: ActorSystem ) {
      implicit val system = actorSystem
      def subscribe( ... ): Unit = {
        actor(new Act{
          // get a reference to an overlay actor here
          ref ! something
          receive = {
            case ...
          }
        })

The problem with this was that whatever I sent to `sender()` from the receiver of the proxy message used to end up in the dead letter queue. It never arrived at my `receive`.

So, in order for the user not having to create an actor himself, I wanted to hand in a class which is capable of communicating with the system but can be used outside of the actor system itself. That's the reason for this construct. I don't like it myself, I would like to change it but the documentation on how to communicate with an actor system from outside of it is rather sparse. From what I see, there is no way to get a reference to a class behind the `ActorRef`.

Hope this explains.